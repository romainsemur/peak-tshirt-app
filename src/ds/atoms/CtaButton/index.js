import { Button } from "@mui/material";

/**
 *
 * @example
 * import {CtaButton} from "ds/atoms";
 * <CtaButton startIcon={<AddShoppingCart /} fullWidth>Add to cart</CtaButton>
 */
export default function CtaButton({ children, fullWidth, startIcon }) {
  return (
    <Button
      variant="contained"
      color="success"
      size="large"
      fullWidth={fullWidth}
      startIcon={startIcon}
    >
      {children}
    </Button>
  );
}
