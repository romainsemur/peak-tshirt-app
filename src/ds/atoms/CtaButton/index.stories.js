import CtaButton from ".";

export default {
  title: "CtaButton",
  component: CtaButton,
};

const Template = (args) => <CtaButton {...args} />;

export const MainUsage = Template.bind({});
MainUsage.args = {
  children: "Click Me",
};
