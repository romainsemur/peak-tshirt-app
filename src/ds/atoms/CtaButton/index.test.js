import { render } from "@testing-library/react";
import CtaButton from ".";

describe("CtaButton", () => {
  test("should pass children and all the other props on MUI Button", () => {
    const { container } = render(
      <CtaButton data-foo="bar">Click Me</CtaButton>
    );
    expect(container).toMatchSnapshot();
  });
});
