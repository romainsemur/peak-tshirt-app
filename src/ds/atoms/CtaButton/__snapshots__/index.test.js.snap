// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CtaButton should pass children and all the other props on MUI Button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeLarge MuiButton-containedSizeLarge css-mkazzp-MuiButtonBase-root-MuiButton-root"
    data-foo="bar"
    tabindex="0"
    type="button"
  >
    Click Me
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
